python/lsst/cp/pipe/ptc.py

```diff
@@ -928,7 +848,7 @@ def _getInitialGoodPoints(means, variances, maxDeviationPositive, maxDeviationNe
 
     def _makeZeroSafe(self, array, warn=True, substituteValue=1e-9):
         """"""
-        nBad = Counter(array)[0]
+        nBad = Counter(np.ravel(np.array(array)))[0]
```

Comment:
Counter is a dict like object with keys that are the elements placed into it and values of the number of times it was added. `nBad` will never be equal to zero just like a dict wont be.

Reply:
Do you have unit tests for some of this behavior? (Of course not every little line needs tested, but just curious how much testing there is)

Reply:
I added some unit tests for this function (and modified it slightly; after fixing a bug).